### ISSUE TITLE
org-tags-search hangs on a fresh 'develop branch' install

### ISSUE THREAD

**Issue by lyndhurst** [syl20bnr/spacemacs#12509]
#### Description :octocat:

I am new to spacemacs, this is my first config, I might be missing something obvious.
I have been using spacemacs for two weeks without any other issues, and spent many hours trying to solve this one. I did not find anything related except this [post related to emacs.](https://emacs.stackexchange.com/questions/48505/help-debugging-org-mode-hangs-on-agenda-tag-search/51459#51459)

I have tried reinstalling everything, reproducing with a config as minimal as possible, nothing helped. By minimal, I mean only 'org', enabled in packages, and no special config like alist, tag faces or todo keyword.

I tried with only one file added to org-agenda-files, with tags only defined at the file level, with a minimal global tag alist as well; everything I could think of. I have no agenda custom commands set yet.

The only way I do have the agenda window opening, without emacs freezing, is without any files added to org-agenda. But as expected, I get no match in that window.

I have to add that all other agenda views are working properly: 'ALL TODOs', Buffer only, By TODO keywords. Also, agenda regexp search (/)  also works properly if I enter one of my tags as a search criteria.

#### Reproduction guide :beetle:
- Start Emacs
- Open an org file with tags
  - file can be in org-agenda-files already
  - or added manually with 'C-c [' : same outcome)
  - headlines can be folded or not: same outcome
- Perform org tag search any of the following ways:
  - ', a m'
  - or ', a M'
  - or 'SPC SPC org-tags-search'
  - or simply click on a tag

*Observed behaviour:* :eyes: :broken_heart:
Spacemacs freezes and CPU increases to 100% of emacs allocated resources.
At that point, no spacemacs commands are responding except C-g.
I cannot interact with the application window either, only kill the process at the os level.
Performing C-g command leaves an empty agenda window.

*Expected behaviour:* :heart: :smile:
No CPU significant increase, and a nice list of headlines matching the selected tag in agenda window.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 7c3d21172)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm better-defaults emacs-lisp org finance themes-megapack)
```

- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11
_I left this part default, as I am not sure how to get it. I am on a kubuntu 18.04 install if this helps._

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (error "No Org agenda currently displayed")
  signal(error ("No Org agenda currently displayed"))
  error("No Org agenda currently displayed")
  org-agenda-check-type(nil agenda)

```

**Comment by lyndhurst** [comment]
I am not sure if I should close the issue. This behavior has been consistent for two weeks on two different installs. I have been practicing this morning, behavior was still the same.
It is now working as expected ! 
The only thing I have changed in my config is the addition of org-refile-targets variable below the org-agenda-files .
Please let me know if I should close the issue myself. I am happy it is working, but puzzled at the same time.

**Comment by lyndhurst** [comment]
I have read a few posts with related issues on tag searches, and it seems that setting the refile targets is relevant to the problem I was having. 
Maybe it is a bug, maybe I got it wrong and it is unrelated, but my searches are now working consistently, and trying to remove my refile targets variable does help me reproduce the issue.
Since I got it working, I will close this issue not to waste any contributor's time.
